bgpd/bgp_evpn_vty.c

```diff
@@ -1060,6 +1060,9 @@ static int bgp_show_ethernet_vpn(struct vty *vty, struct prefix_rd *prd,
 
 				json_object_int_add(json_prefix_info,
 					"prefixLen", rm->p.prefixlen);
+
+				if (rd_header && use_json)
```

Comment:
This is already under use_json check. So just checking rd_header would be sufficient.

Also, in the scenario that you had explained rd_header will never set to Zero and so we will allocate multiple json_nroute objects.

Reply:
yep fixing